numpyro/primitives.py

```diff
@@ -151,13 +153,21 @@ def param(name, init_value=None, **kwargs):
     """
     # if there are no active Messengers, we just draw a sample and return it as expected:
     if not _PYRO_STACK:
+        assert not callable(init_value), \
+            "A callable init_value needs to be put inside a numpyro.handlers.seed handler."
         return init_value
 
+    if callable(init_value):
+        def fn(init_fn, *args, **kwargs):
+            return init_fn(prng_key())
+    else:
+        fn = identity
```

Comment:
LGTM. For my own understanding, do we use kwargs to pass in constraints?

Reply:
Yes, in svi we will use kwargs from param node in model trace to decide if we need to transform the value.